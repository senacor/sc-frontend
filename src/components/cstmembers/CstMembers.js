import React, { Component } from 'react';
import { connect } from 'react-redux';
import Avatar from '@material-ui/core/Avatar';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import * as actions from '../../actions';
import { formatDateForFrontend } from '../../helper/date';

class CstMembers extends Component {
  componentDidMount() {
    this.props.getCstMembers();
  }

  render() {
    const { cstMembers } = this.props;

    return (
      <List>
        {cstMembers &&
          cstMembers.map(cstMember => {
            return (
              <ListItem key={cstMember.id}>
                <Avatar>{`${cstMember.firstName.charAt(
                  0
                )}${cstMember.lastName.charAt(0)}`}</Avatar>
                <ListItemText
                  primary={`${cstMember.firstName} ${cstMember.lastName}`}
                  secondary={`Datum letztes PR: ${formatDateForFrontend(
                    cstMember.dateOfLastPr
                  )}`}
                />
              </ListItem>
            );
          })}
      </List>
    );
  }
}

export default connect(
  state => ({
    cstMembers: state.cstMembers
  }),
  {
    getCstMembers: actions.getCstMembers
  }
)(CstMembers);
