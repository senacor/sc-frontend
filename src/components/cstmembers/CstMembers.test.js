import React from 'react';
import CstMembers from './CstMembers';

describe('CstMembers component', function() {
  it('should match snapshot', () => {
    let cut = <CstMembers />;

    expect(cut).toMatchSnapshot();
  });
});
