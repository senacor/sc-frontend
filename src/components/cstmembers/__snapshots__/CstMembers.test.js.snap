// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CstMembers component should match snapshot 1`] = `<Connect(CstMembers) />`;
